cpp/src/arrow/tensor-test.cc

```diff
@@ -155,6 +155,49 @@ TEST(TestTensor, CountNonZeroForNonContiguousTensor) {
   AssertCountNonZero(t, 8);
 }
 
+TEST(TestTensor, Equals) {
```

Comment:
Is it possible to also test unequal tensors? Contiguous, non-contiguous...

Reply:
Oh, I forgot to write unequal cases.
I added them at d40c3c7745a318507f05f174ae9dac5a1d381718